applications/ConstitutiveLawsApplication/custom_constitutive/constitutive_laws_integrators/high_cycle_fatigue_law_integrator.h

```diff
@@ -173,10 +173,24 @@ class HighCycleFatigueLawIntegrator
                                             double& rN_f)
 	{
         const Vector& r_fatigue_coefficients = rMaterialParameters[HIGH_CYCLE_FATIGUE_COEFFICIENTS];
-        const double yield_stress = rMaterialParameters.Has(YIELD_STRESS) ? rMaterialParameters[YIELD_STRESS] : rMaterialParameters[YIELD_STRESS_TENSION];
+        double ultimate_stress = rMaterialParameters.Has(YIELD_STRESS) ? rMaterialParameters[YIELD_STRESS] : rMaterialParameters[YIELD_STRESS_TENSION];
+        const double yield_stress = ultimate_stress;
+
+        // The calculation is prepared to update the rN_f value when using a softening curve which initiates with hardening.
+        // The jump in the advance in time process is done in these cases to the Syield rather to Sult.
+        const int softening_type = rMaterialParameters[SOFTENING_TYPE];
+        if (softening_type == 3) {
```

Comment:
I think that ou can use the enum, look how we did it in plasticity or even damage, easy to adapt

Reply:
Done. Thanks for the comment